library/panic_abort/src/lib.rs

```diff
@@ -59,7 +59,18 @@ pub unsafe extern "C" fn __rust_start_panic(_payload: usize) -> u32 {
             //
             // Note: this is the same implementation as in libstd's `abort_internal`
             unsafe fn abort() -> ! {
-                llvm_asm!("int $$0x29" :: "{ecx}"(7) ::: volatile); // 7 is FAST_FAIL_FATAL_APP_EXIT
+                const FAST_FAIL_FATAL_APP_EXIT: usize = 7;
+                cfg_if::cfg_if! {
+                    if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
+                        asm!("int $$0x29", in("ecx") FAST_FAIL_FATAL_APP_EXIT);
+                    } else if #[cfg(target_arch = "arm")] {
+                        asm!(".inst 0xDEFB", in("r0") FAST_FAIL_FATAL_APP_EXIT);
```

Comment:
I'm not sure if this applies to Windows, but you should check for `cfg(target_feature = "thumb_mode")` so that you can generate the correct instruction in ARM and Thumb states which have different encodings. `0xdefb` is only a valid encoding in thumb state.

Reply:
I'm certainly no expert here, but I believe that there is only one `arm` Windows target `thumbv7a-pc-windows-msvc` so effectively this branch will only be triggered on targets with "thumb-mode", but in the interest of things not breaking in the future, it seems like a good idea to make this change. I will push a new commit with this.